usaspending_api/accounts/v2/views/budget_function.py

```diff
@@ -0,0 +1,48 @@
+from django.db.models import Q
+from rest_framework.response import Response
+
+from usaspending_api.accounts.models import TreasuryAppropriationAccount
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.views import APIDocumentationView
+
+
+class ListBudgetFunctionViewSet(APIDocumentationView):
+    """
+    This route sends a request to the backend to retrieve all Budget Functions associated with a TAS, ordered by Budget
+        Function code.
+    endpoint_doc: /budget_function.md
+    """
+    @cache_response()
+    def get(self, request):
+        # Retrieve all Budget Functions, grouped by code and title, ordered by code
+        results = TreasuryAppropriationAccount.objects \
+            .filter(~Q(budget_function_code=''), ~Q(budget_function_code=None)) \
+            .values('budget_function_code', 'budget_function_title') \
+            .order_by('budget_function_code').distinct()
+
+        return Response({'results': results})
+
+
+class ListBudgetSubfunctionViewSet(APIDocumentationView):
+    """
+    This route sends a request to the backend to retrieve all Budget Subfunctions associated with a TAS, ordered by
+        Budget Subfunction code. Can be filtered by Budget Function.
+    endpoint_doc: /budget_subfunction.md
+    """
+    @cache_response()
+    def post(self, request):
+        # Retrieve all Budget Subfunctions
+        queryset = TreasuryAppropriationAccount.objects \
+            .filter(~Q(budget_subfunction_code=''), ~Q(budget_subfunction_code=None))
```

Comment:
See other comment.

Reply:
Testing between both implementations, they were about the same in performance. Moving on.